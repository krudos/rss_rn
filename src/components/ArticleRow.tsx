import {FeedItem} from 'react-native-rss-parser';
import React, {FC, useCallback} from 'react';
import {
  Alert,
  Dimensions,
  Linking,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import AutoHeightWebView from 'react-native-autoheight-webview';

interface PostProps {
  data: FeedItem;
}
export const ArticleRow: FC<PostProps> = ({data}) => {
  const showUrl = useCallback(async () => {
    const url = data.links[0].url;
    const supported = await Linking.canOpenURL(url);
    if (supported) {
      await Linking.openURL(url);
    } else {
      Alert.alert(`Don't know how to open this URL: ${url}`);
    }
  }, [data.links]);
  return (
    <View key={data.id}>
      <Text>{data.title}</Text>
      <Text>{data.published}</Text>

      <TouchableOpacity key={data.id} onPress={showUrl} style={styles.button}>
        <Text style={styles.buttonText}>Open</Text>
      </TouchableOpacity>
      <AutoHeightWebView
        style={styles.webView}
        originWhitelist={['*']}
        source={{html: data.description}}
        scalesPageToFit={true}
        viewportContent={'width=device-width, user-scalable=no'}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  topContainer: {flex: 1, alignItems: 'center'},
  bottomContainer: {},
  background: {
    flex: 1,
  },
  image: {height: 120},
  button: {marginHorizontal: 40, backgroundColor: 'blue'},
  buttonText: {
    fontSize: 24,
    fontWeight: '300',
    color: 'white',
    textAlign: 'center',
  },
  webView: {width: Dimensions.get('window').width - 15, marginTop: 35},
});
