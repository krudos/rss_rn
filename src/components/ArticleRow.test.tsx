import React from 'react';
import {render} from '@testing-library/react-native';
import {ArticleRow} from './ArticleRow';
import {FeedItem} from 'react-native-rss-parser';
describe('ArticleItem', () => {
  test('renders correctly', () => {
    // @ts-ignore
    const mock: FeedItem = {
      content: '',
      description: '',
      itunes: {
        authors: undefined,
        block: undefined,
        duration: '',
        explicit: '',
        image: undefined,
        isClosedCaptioned: undefined,
        order: undefined,
        subtitle: '',
        summary: undefined,
      },
      links: [{url: '', rel: ''}],
      published: '',
      id: '123',
      title: 'test',
    };
    const {toJSON} = render(<ArticleRow data={mock} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
