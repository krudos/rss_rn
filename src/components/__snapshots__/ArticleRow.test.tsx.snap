// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleItem renders correctly 1`] = `
<View>
  <Text>
    test
  </Text>
  <Text>
    
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "blue",
        "marginHorizontal": 40,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 24,
          "fontWeight": "300",
          "textAlign": "center",
        }
      }
    >
      Open
    </Text>
  </View>
  <Webview
    originWhitelist={
      Array [
        "*",
      ]
    }
    scalesPageToFit={true}
    source={
      Object {
        "html": "",
      }
    }
    style={
      Object {
        "marginTop": 35,
        "width": 735,
      }
    }
    viewportContent="width=device-width, user-scalable=no"
  />
</View>
`;
